File: src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceS.scala
package nu.tengstrand.tetrisanalyzer.piece

object PieceS {
  def apply() = new PieceS
}

class PieceS extends Piece {
  val number = 3.toByte
  val character = 'S'
  protected val widths = Array(3, 2)
  protected val heights = Array(2, 3)
  protected val shapes = Array(
    new PieceShape(Array(Point(1,0), Point(2,0), Point(0,1), Point(1,1))),
    new PieceShape(Array(Point(0,0), Point(0,1), Point(1,1), Point(1,2)))
  )
}

File: langs/java/src/main/java/com/github/tetrisanalyzer/piece/PieceS.java
package com.github.tetrisanalyzer.piece;

public class PieceS extends Piece {
    @Override public byte number() { return 3; }
    @Override public char character() { return 'S'; }
    @Override protected int[] widths() { return new int[] { 3, 2 }; }
    @Override protected int[] heights() { return new int[] { 2, 3 }; }
    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(1,0), new Point(2,0), new Point(0,1), new Point(1,1)),
            new PieceShape(new Point(0,0), new Point(0,1), new Point(1,1), new Point(1,2))
        };
    }
}
